File: gaia-sdk-python/gaia_sdk/graphql/request/type/UpdatedTenantImpulse.py

from gaia_sdk.graphql.request.type.Tenant import Tenant

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class UpdatedTenantImpulse(list):
    """
    Impulse which indicates the result of a update Tenant impulse
    """

    def id(self):
        self.append(lambda x: "id")

    """
    the Tenant instance
    """
    def data(self, config: Callable[['Tenant'], None]):
        def callback(registry: VariableRegistry):
            entity = Tenant()
            config(entity)
            return "data {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))

File: gaia-sdk-javascript/src/graphql/request/type/UpdatedTenantImpulse.ts

import {Tenant} from "./Tenant";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Impulse which indicates the result of a update Tenant impulse
 */
export class UpdatedTenantImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "UpdatedTenantImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    /**
     * the Tenant instance
     */
    public data = (config: (_:Tenant) => void) => this.push((registry) => {
        const entity = new Tenant();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

